pkg/adaptor/xds/v3/route.go

```diff
@@ -112,10 +112,27 @@ func (adaptor *adaptor) translateVirtualHost(prefix string, vhost *routev3.Virtu
 			Hosts:      hosts,
 			Uris:       []string{uri},
 			UpstreamId: id.GenID(cluster),
-			Vars:       vars,
 		}
+		if len(vars) > 0 {
+			r.Vars = vars
+		}
+
+		plugin, err := adaptor.translateRouteAction(route)
+		if err != nil {
+			return nil, err
+		}
+		if plugin != nil {
+			if len(vars) > 0 && plugin.TrafficSplit != nil {
+				plugin.TrafficSplit.Rules[0].Match = vars
```

Comment:
We don't need to assign the `vars` to the `traffic-split` plugin as `vars` is the essential Route match criterion.